### PR TITLE
[PHEE-637] Batch Payment Mock API's

### PR DESCRIPTION
**Pull request by abhinavm117**
## Description

* PR title should have jira ticket enclosed in `[]`.<br>
Format: ``` [jira_ticket] description```<br>
ex: [phee-123] PR title.
* [Jira ticket](https://fynarfin.atlassian.net/browse/PHEE-637?atlOrigin=eyJpIjoiNmYxODc3YTFmMDhhNGU3NjlhOTE2ZWIwOGM4MzgwZjUiLCJwIjoiamlyYS1zbGFjay1pbnQifQ)
* Describe the changes made and why they were made.
Postman collection PR -  https://github.com/openMF/ph-ee-env-template/pull/411
## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!
- [ ] Followed the PR title naming convention mentioned above.

- [ ] Design related bullet points or design document link related to this PR added in the description above.

- [ ] Updated corresponding Postman Collection or Api documentation for the changes in this PR.

- [ ] Created/updated unit or integration tests for verifying the changes made.

- [ ] Added required Swagger annotation and update API documentation with details of any API changes if applicable

- [ ] Followed the naming conventions as given in https://docs.google.com/document/d/1Q4vaMSzrTxxh9TS0RILuNkSkYCxotuYk1Xe0CMIkkCU/edit?usp=sharing

### CODE REVIEW

File: src/main/java/org/mifos/connector/mockpaymentschema/api/BatchApi.java [hunk 1]
```diff
@@ -41,4 +45,15 @@ public ResponseEntity<Object> getAuthorization(@PathVariable String batchId,
         }
         return new ResponseEntity<>(HttpStatus.ACCEPTED);
     }
+
+    @GetMapping(value = "/batches/{batchId}/summary", produces = MediaType.APPLICATION_JSON_VALUE)
+    public BatchDTO batchSummary(@PathVariable String batchId) {
+        return batchService.getBatchSummary(batchId);
+    }
+
+    @GetMapping(value = "/batches/{batchId}/detail", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mockpaymentschema/api/BatchApi.java:54]: This API documentation should be upated. Please attach link to postman PR.
Also make sure that the validations and error handling is done.

ref: https://docs.google.com/spreadsheets/d/13TdoMUXtmV4MqwafbZu4_zvwOUhbO2WTs1UxZOLXHXI/edit#gid=0

**Review by fynmanoj — CHANGES_REQUESTED** [review]
Requested clarification

File: src/main/java/org/mifos/connector/mockpaymentschema/service/BatchService.java [hunk 2]
```diff
@@ -42,4 +57,68 @@ public void getAuthorization(String batchId, String clientCorrelationId, Authori
...
+                        TransferStatus.IN_PROGRESS, batchId);
+            }
+            transactionList.add(transfer);
+        }
+        return transactionList;
+    }
+
+    private Transfer getSingleTransaction(int index, Long workflowInstanceKey, String requestId, TransferStatus status, String batchId) {
+        String id = String.valueOf(index);
+        Date startedAt = new Date(1685536200000L);
```
Comment (fynmanoj) [inline src/main/java/org/mifos/connector/mockpaymentschema/service/BatchService.java:112]: is there a test case available for this?

**Review by fynmanoj — CHANGES_REQUESTED** [review]

**Comment by fynmanoj** [comment]
Please add description and title for this PR

File: src/main/java/org/mifos/connector/mockpaymentschema/service/BatchService.java (hunk 2 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/connector/mockpaymentschema/service/BatchService.java:112]: No, manoj. All data is mocked.

File: src/main/java/org/mifos/connector/mockpaymentschema/api/BatchApi.java (hunk 1 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/connector/mockpaymentschema/api/BatchApi.java:54]: https://github.com/openMF/ph-ee-env-template/pull/411
I have added postman collection for this.

File: src/main/java/org/mifos/connector/mockpaymentschema/service/BatchService.java (hunk 2 above)
Comment (abhinavm117) [reply src/main/java/org/mifos/connector/mockpaymentschema/service/BatchService.java:112]: No, it will always send same data. No need for test case here

**Pull request merged**